pillowtop/listener.py

```diff
@@ -399,6 +399,37 @@ def run(self):
         super(PythonPillow, self).run()
 
 
+class MigrationPillowMixin(object):
+    """
+    The migration pillow is intended to be used for migrating couchdocs from one couch database to another.
+    This should be used in combination of the couch_migrate command to initially seed the destination
+    database:
+
+    https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cleanup/management/commands/couch_migrate.py
+
+    The pillow ensures that all documents that get deleted in the source database are subsequently updated in
+    the destination database.
+    """
+
+    # Override this to determine what constitutes a deleted document
+    is_deleted = lambda doc: False
+
+    # Override this function to get the db instance you'd like to migrate to
+    dest_db = lambda: None
```

Comment:
for the device logs, it would look something like this:
```
lambda: DeviceLogXFormInstance.get_db()
```
we may be able to get away with just defining the destination class, though i thought leaving it a function would make it more flexible